internal/memhttp/internal/config.go

```diff
@@ -0,0 +1,35 @@
+// Copyright 2021-2023 The Connect Authors
```

Comment:
This package would seem a tad overkill to me - is there a reason to not just make the types within here private to `internal/memhttp`?

Reply:
@jhump has a plan for restricting the `WithCleanupTimeout` to only the `memhttptest` package: https://github.com/connectrpc/connect-go/pull/594#discussion_r1353458824

Reply:
Let's move it back to `memhttp` for now and simplify. I can propose an alternate formulation later and see what folks think of it.